content/vault/v1.21.x (rc)/content/api-docs/secret/identity/mfa/totp.mdx

```diff
@@ -300,3 +303,42 @@ $ curl \
     --data @payload.json \
     http://127.0.0.1:8200/v1/identity/mfa/method/totp/admin-destroy
 ```
+
+### Self-enroll in TOTP MFA <EnterpriseAlert product="vault" inline />
+
+This endpoint allows a user to self-enroll in TOTP MFA by generating their TOTP
+secret (QR code) during login if they do not already have one. This endpoint is
+only available in Vault Enterprise if the `enable_self_enrollment` parameter is
+set to true on the TOTP MFA method.
+
+| Method | Path                                    |
+|:-------|:----------------------------------------|
+| `POST` | `/identity/mfa/method/totp/self-enroll` |
```

Comment:
Do you think it's worth mentioning somewhere that the UI only enters the self-enroll workflow if only one enforcement is configured because that's when `/validate` returns `self_enrollment_enabled: true`? Maybe in the section [Configure the TOTP Method](https://github.com/hashicorp/web-unified-docs/pull/1041/files#diff-55f9708b6a66d33262ac641fc09e937e96532b6615b53bcc4545332386eb3407R249). 

Technically, users could circumvent this by manually requesting `self-enroll` via curl, right? Though, I'm not sure it's worth going into that level of detail... 

@kubawi

Reply:
Very good point, I completely forgot about that. Yes, I think we should mention it wherever we can as long as this makes sense, to avoid users being confused. Maybe it would make sense to add screenshots of the whole flow somewhere? 🤔

Reply:
Yeah - I was holding off on screenshots until I heard back from the education team (since I just took a best guess at where to add docs 😂 )

But yes - @schavis - do you have thoughts on:

1. If and where we should add screenshots to walkthrough the UI login flow? 
2. Where we should callout that even if `enable_self_enrollment` is configured to `true` the UI will only enter self-enroll if only one MFA login enforcement is configured.

Reply:
I'm not sure what happened to my original reply from last night =\

> If and where we should add screenshots to walkthrough the UI login flow?

In general, we try to limit the number of screenshots due to the maintenance burden. Is there a "hero" screenshot we can reference to help folks orient themselves in the GUI?

> Where we should callout that even if enable_self_enrollment is configured to true 

Since we're essentially calling out a configuration gotcha, I'd recommend adding it as part of the instructions under "Option 1: Configure TOTP for self-enrollment".